From: "Addisu Z. Taddese" <addisu@openrobotics.org>
Date: Fri, 29 Mar 2024 15:00:46 -0500
Subject: [PATCH] Fix pkgconfig installation so it works with
 `CMAKE_STAGING_PREFIX`

Signed-off-by: Addisu Z. Taddese <addisu@openrobotics.org>
---

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f3fe251..df820b0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -529,13 +529,14 @@ if(ABSOLUTE_CMAKE_INSTALL_LIBDIR)
 else()
   # When "CMAKE_INSTALL_*DIR" are relative, we should ensure the installed
   # files contain relocatable references to install paths.
-  set(PC_CONFIG_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+  set(PC_CONFIG_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+  set(PC_CONFIG_INSTALL_ABS_DIR ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/pkgconfig)
   set(PC_LIBDIR "\${prefix}/${CMAKE_INSTALL_LIBDIR}")
   set(PC_INCLUDEDIR "\${prefix}/${CMAKE_INSTALL_INCLUDEDIR}")
 endif()
 file(RELATIVE_PATH
   RELATIVE_PATH_TO_INSTALL_PREFIX
-  "${PC_CONFIG_INSTALL_DIR}"
+  "${PC_CONFIG_INSTALL_ABS_DIR}"
   "${CMAKE_INSTALL_PREFIX}"
 )
 if(DART_VERBOSE)
